app/parse.py

```diff
@@ -1,12 +1,53 @@
 from dataclasses import dataclass
 
+import requests
+from bs4 import Tag, BeautifulSoup
+
+
+MATE_URL = "https://mate.academy/"
+
 
 @dataclass
 class Course:
     name: str
     short_description: str
     duration: str
+    modules: int = None
+    topics: int = None
+
+
+def get_detail_course_info(course_detail_url: str) -> tuple[int, int]:
+    detail_page = requests.get(MATE_URL + course_detail_url[1:]).content
+    soup = BeautifulSoup(detail_page, "html.parser")
+
+    program = soup.select_one(
+        "section.profession-page-container > "
+        "div.CourseProgram_content__tvxzY"
+    )
+    modules = len(program.select("div.CourseProgram_modules__GA_PJ > ul > li"))
+    topics = int(program.select_one(".FactBlock_factNumber__d_8nn").text)
```

Comment:
Similar to the previous line, this line assumes that the `select_one` method will always find an element and that the text can be converted to an integer. Consider adding error handling for cases where the element is not found or the text is not a valid integer.